mteb/models/model_implementations/eagerworks_models.py

```diff
@@ -0,0 +1,334 @@
+from typing import Any
+
+import torch
+from PIL import Image
+from torch.utils.data import DataLoader
+from tqdm.auto import tqdm
+
+from mteb._requires_package import (
+    requires_image_dependencies,
+    requires_package,
+)
+from mteb.abstasks.task_metadata import TaskMetadata
+from mteb.models.abs_encoder import AbsEncoder
+from mteb.models.model_meta import ModelMeta, ScoringFunction
+from mteb.models.models_protocols import PromptType
+from mteb.types import Array, BatchedInput
+
+
+class EagerEmbedV1Wrapper(AbsEncoder):
+    """Wrapper for EagerEmbed single-vector embedding models."""
+
+    def __init__(
+        self,
+        model_name: str,
+        revision: str | None = None,
+        device: str | None = None,
+        image_size: int = 784,
+        use_peft: bool = False,
+        **kwargs,
+    ):
+        requires_image_dependencies()
+        requires_package(
+            self, "peft", model_name, "pip install mteb[peft]"
+        )
+        requires_package(
+            self, "qwen_vl_utils", model_name, "pip install mteb[qwen_vl_utils]"
+        )
```

Comment:
I would convert the optional dependencies to be model specific (so one group installs everything needed)